cve_bin_tool/extractor.py

```diff
@@ -109,6 +113,30 @@ async def extract_file_rpm(self, filename, extraction_path):
                     return 1
         return 0
 
+    async def extract_file_zst(self, filename, extraction_path):
+        """Extract zstd compressed files"""
+
+        dctx = zstandard.ZstdDecompressor()
+        with ErrorHandler(mode=ErrorMode.Ignore) as e:
+            with open(filename, "rb") as f:
+                with dctx.stream_reader(f) as reader:
+                    with tarfile.open(mode="r|", fileobj=reader) as tar:
+                        tar.extractall(extraction_path)
+        return e.exit_code
+
+    async def extract_file_pkg(self, filename, extraction_path):
+        """Extract pkg files"""
+        if not await aio_inpath("tar"):
+            with ErrorHandler(mode=self.error_mode, logger=self.logger):
+                raise Exception("tar is required to extract .pkg files")
+        else:
+            stdout, stderr, return_code = await aio_run_command(
+                ["tar", "xf", filename, "-C", extraction_path]
```

Comment:
Can this be replaced by `tarfile`? If `tarfile` is a slower approach then let's not do it.